service-config/src/main/java/com/palantir/config/service/ServiceDiscoveryConfiguration.java

```diff
@@ -136,6 +136,12 @@
         return getServiceConfig(serviceName).readTimeout();
     }
 
+    public final boolean isServiceEnabled(String serviceName) {
+        ServiceConfiguration serviceConfig = getServices().get(serviceName);
+
+        return serviceConfig != null && !serviceConfig.uris().isEmpty();
```

Comment:
The main reason I want to add this function was because the none-empty check. We need to support services with empty list of Uris - this happens when service discovery tries to discover the service and finds nothing. So technically those services aren't enabled but will have an entry in the map. 

We could also find a way to filter services - ignore and remove those from the services list somehow (probably from the service discovery constructor). This seems a bit easier to understand but the code might be messier.


> On Apr 29, 2016, at 12:51 AM, Robert Fink <notifications@github.com> wrote:
> 
> In service-config/src/main/java/com/palantir/config/service/ServiceDiscoveryConfiguration.java:
> 
> > @@ -136,6 +136,12 @@
> >          return getServiceConfig(serviceName).readTimeout();
> >      }
> >  
> > +    public final boolean isServiceEnabled(String serviceName) {
> > +        ServiceConfiguration serviceConfig = getServices().get(serviceName);
> > +
> > +        return serviceConfig != null && !serviceConfig.uris().isEmpty();
> Would it make sense to force uris() to be non-empty in the constructor instead?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>